src/compiler/transformers/declarations/diagnostics.ts

```diff
@@ -345,7 +345,7 @@ namespace ts {
 
                 case SyntaxKind.MethodDeclaration:
                 case SyntaxKind.MethodSignature:
-                    if (hasModifier(node.parent, ModifierFlags.Static)) {
+                    if (hasEffectiveModifier(node.parent, ModifierFlags.Static)) {

```

Comment:
should this be syntactic like the others in this file? I didn't think that non-syntactic `static` marking was possible.

Reply:
It should be, but it doesn't matter too much here. "effective" modifiers include syntactic modifiers, though I've tried to limit the cases where I call `hasEffectiveModifier` et al when there's no need to do the extra computation for jsdoc modifiers. I'll fix this case.